lib/IRGen/GenClangType.cpp

```diff
@@ -220,6 +220,9 @@ clang::CanQualType GenClangType::visitStructType(CanStructType type) {
   CHECK_NAMED_TYPE(swiftDecl->getASTContext().getSwiftName(
                      KnownFoundationEntity::NSZone),
                    ctx.VoidPtrTy);
+  CHECK_NAMED_TYPE("WindowsBOOL",
+                   ctx.getCanonicalType(ctx.getTypedefType(
+                       ctx.buildImplicitTypedef(ctx.IntTy, "BOOL"))));
```

Comment:
This only cares about canonical (i.e. unsugared) types, so you should just use `ctx.IntTy`.

Reply:
Oh, awesome, that makes this significantly simpler.